src/huggingface_hub/repository.py

```diff
@@ -29,20 +29,27 @@ def __init__(
         is_done_method: Callable,
         status_method: Callable,
         process: subprocess.Popen,
+        post_method: Optional[Callable] = None,
     ):
         self.title = title
         self._is_done = is_done_method
         self._status = status_method
         self._process = process
         self._stderr = ""
         self._stdout = ""
+        self._post_method = post_method
 
     @property
     def is_done(self) -> bool:
         """
         Whether the process is done.
         """
-        return self._is_done()
+        result = self._is_done()
+
+        if result and self._post_method is not None:
+            self._post_method()
```

Comment:
Very smart!

Reply:
I don't know the code too well, but it's a bit weird to me that a property "is this command done" now calls a `"post"` method. Wouldn't it be cleaner to separate the `self._post_method()` from `is_done`?

Reply:
In that case the user will need to manually check when an async git commit is finished to call the post method, which is what we are trying to avoid.

Reply:
I understand what you mean Patrick, let me give you some context: we're leveraging the `subprocess.Popen` approach to launch the process in a non-blocking manner. This approach means we're spawning a subprocess and keeping track of it in the `CommandInProgress` utility.

The process does not have the ability to let the Python runtime know when it's done: it's up to the Python runtime to check when it is done by checking its `status`.

Therefore, here, we have three options:
1. Check the status regularly and when it is done run the `git lfs prune`
2. Let the user check the status, and when it is done run the `git lfs prune`
3. Don't do any of that and let the user handle the `git lfs prune` when they feel like it

We've chosen to do 2., as that's the least intensive way of doing things; but approach 1. might be better as the `git lfs prune` might take a bit of time for larger files, which is currently blocking when the user checks for `is_done`. For now I'm going to add a log message when it is pruning, but I'm open to switching from 2. to 1. if you feel strongly.

3. is not a great user API.

Reply:
Good for me!